### PR TITLE
Add 'resolver' function wasm API

### PR DESCRIPTION
**Pull request by mykmartin**
This allows particles to specify URLs like 'https://$arcs/...'

### CODE REVIEW

File: src/runtime/loader.ts [hunk 1]
```diff
@@ -68,14 +68,23 @@ export class Loader {
     return this.loadFile(file, 'utf-8') as Promise<string>;
   }
 
-  async loadBinary(file: string): Promise<ArrayBuffer> {
-    if (/^https?:\/\//.test(file)) {
-      return fetch(file).then(res => res.arrayBuffer());
+  async loadWasmBinary(spec): Promise<ArrayBuffer> {
+    // TODO: use spec.implBlobUrl if present?
+    this.mapParticleUrl(spec.implFile);
```
Comment (alxrsngrtn) [inline src/runtime/loader.ts]: random idea:
```
const mapped = this.mapParticleUrl(spec.implFile);
...
mapParticleUrl(path: string) {
  return path;
}```

File: src/runtime/wasm.ts [hunk 2]
```diff
@@ -457,6 +464,11 @@ export class WasmContainer {
     this.particleMap.set(innerParticle, particle);
   }
 
+  // Allocates memory in the wasm container; the calling particle is responsible for freeing.
+  resolve(urlPtr: WasmAddress) {
```
Comment (alxrsngrtn) [inline src/runtime/wasm.ts]: Let's add the return type: `: WasmAddress`

Comment (mykmartin) [reply src/runtime/wasm.ts]: Done.

**Comment by mykmartin** [comment]
> `pec-industry-web.ts`, line 17: need to change `loader._resolve` to `loader.resolve`.

Good catch; fixed.

File: src/runtime/loader.ts (hunk 1 above)
Comment (mykmartin) [reply src/runtime/loader.ts]: `mapParticleUrl` doesn't return anything - it's adding to the internal url map to allow `$` macros based on the particle name and location.

**Comment by mykmartin** [comment]
And now it seems dismissing Alex's review has nuked the LGTM. Sigh. Adding Josh to stamp this (and we may want to figure out how to use the 'request changes' feature).

**Pull request merged**